OADP-4058: Adding structure for OADP and 3scale user story

apurvabhide17 opened the pull request:
Version(s):
OCP 4.14 - 4.18

Issue:
[OADP-4058](https://issues.redhat.com/browse/OADP-4058)

Link to docs preview:
[Backing up and restoring 3scale using OADP](https://85994--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-and-restoring-3scale-by-using-oadp)

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
Parent document - https://docs.google.com/document/d/1MOy3Vsx9bsxJsCZkWA099NmfB39AHh31w4bjHAuHxaw/edit?tab=t.0#heading=h.b8a53qyys7gh
The changes are approved by QE and SME. Waiting a final peer review before merge-review.

apurvabhide17 commented:
/label OADP

apurvabhide17 commented:
/label peer-review-needed

ShaunaDiaz commented:
@apurvabhide17 All PRs pulled against the `main` branch must also go in 4.18. If this content cannot go in 4.18, you'll want to have it open against 4.17 and picked back.

ShaunaDiaz commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,147 @@
+deployment.apps/threescale-operator-controller-manager-v2 scaled
+----
+
+. Scale down all the DeploymentConfigs:
```
```suggestion
. Scale down all the `DeploymentConfig` objects by creating and running the following script:
```

ShaunaDiaz commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,147 @@
+----
+
+. Scale down all the DeploymentConfigs:
++
```
.. Create the script by using the following example:

ShaunaDiaz commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,147 @@
+    oc scale dc/$dc --replicas=0 -n threescale
+done
+----
++
```
```suggestion
.. Run the script you created by using the following command:
```
Or something like this. It is not one step and one command; we're clearly doing two things here.

mpershina replied:
Agree with "Run the script  you created". But we should drop "by using the following command" :-)

ShaunaDiaz commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,147 @@
++
+[source,terminal]
+----
+$ ./scaledowndc.sh 
```
```suggestion
$ ./scaledowndc.sh 
```
Hmm, is this a valid command?

ShaunaDiaz commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,147 @@
+[source,terminal]
+----
+$ ./scaledowndc.sh 
+
```
```suggestion
+
.Example output
[source,terminal]
----
```

ShaunaDiaz commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,147 @@
+deploymentconfig.apps.openshift.io/zync-que scaled
+----
+
+. Delete the system-mysql DeploymentConfig:
```
```suggestion
. Delete the `system-mysql` `DeploymentConfig` object by running the following command:
```

ShaunaDiaz commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,147 @@
+[source,terminal]
+----
+$ oc delete dc system-mysql -n threescale
+
```
again, separate input from output

ShaunaDiaz commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,147 @@
+deploymentconfig.apps.openshift.io "backend-redis" deleted
+----
+
+. Restore the mysql database using YAML:
```
```suggestion
. Restore the `mysql` database with a YAML file by using the following command:
```

ShaunaDiaz commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,147 @@
+apiVersion: velero.io/v1
+kind: Restore
+metadata:
+  name: restore-mysql
```
Should the `name` here match the preceding command, or the `backupName` below?

ShaunaDiaz commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,147 @@
+
+.Verification 
+
+. Verify if the `PodVolumeRestore` is completed:
```
```suggestion
. Verify that the `PodVolumeRestore` is completed by running the following command:
```

ShaunaDiaz commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,147 @@
+[source,terminal]
+----
+$ oc get podvolumerestore -n openshift-adp
+
```
separate input v output

ShaunaDiaz commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,147 @@
+restore-mysql-z7x7l     threescale   system-mysql-2-kjkhl    kopia           example-claim           Completed   380415       380415      40m
+----
+
+. Verify the additional persistent volume claim (PVC) has also been restored:
```
```suggestion
. Verify that the additional persistent volume claim (PVC) has also been restored by running the following command:
```

ShaunaDiaz commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,147 @@
+[source,terminal]
+----
+$ oc get pvc -n threescale
+
```
separate input v output

ShaunaDiaz commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,147 @@
+system-searchd          Bound    pvc-afbf606c-d4a8-4041-8ec6-54c5baf1a3b9   1Gi        RWO            gp3-csi        <unset>                 68m
+----
+
+.Next steps
```
```suggestion
.Next step
```

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
+
+[id="restoring-the-secrets-and-apimanager_{context}"]
+= Restoring the secrets and APIManager
+
```
needs summary sentence

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
+
+.Prerequisites
+
+. You have backed up the 3scale Operator.
```
```suggestion
. You backed up the 3scale Operator.
```

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
+.Prerequisites
+
+. You have backed up the 3scale Operator.
+. You have backed up mysql and Redis databases.
```
```suggestion
. You backed up mysql and Redis databases.
```

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
+
+.Procedure
+
+. Delete the 3scale operator CRDs along with the "threescale" namespace:
```
```suggestion
. Delete the 3scale Operator custom resource definitions (CRDs) along with the "threescale" namespace by running the following command:
```

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
+[source,terminal]
+----
+$ oc delete project threescale
+
```
separate input v output

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
+"threescale" project deleted successfully
+----
++
+NOTE: If restoring on a separate cluster, ensure that {oadp-short} is installed and configured with `nodeAgent` enabled on the destination cluster similar to the source cluster.
```
properly format admonition and determine proper admonition type
```suggestion
[IMPORTANT]
====
If you are restoring the database on a different cluster than you backed up, ensure that {oadp-short} is installed and configured with `nodeAgent` enabled on the destination cluster as it was on to the source cluster.
====
```

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
++
+NOTE: If restoring on a separate cluster, ensure that {oadp-short} is installed and configured with `nodeAgent` enabled on the destination cluster similar to the source cluster.
+
+. Restore the Operator if you backed up the Operator. Otherwise, install a fresh 3scale operator:
```
Which is this operation? And where is the relevant command? This should be two steps, IMHO. Something like, 
. Get the 3scale Operator running by either restoring a backup or installing a new instance:

.. restore step
etc
.. install step
etc

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
+  itemOperationTimeout: 4h0m0s
+----
+
+. Manually create the `s3-credentials` secret: 
```
```suggestion
. Manually create the `s3-credentials` Secret by using the following YAML example: 
```

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
+type: Opaque
+----
+
+. Scale down the 3scale-operator:
```
```suggestion
. Scale down the 3scale Operator by running the following command:
```

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
+  startTimestamp: "2024-09-24T05:54:45Z"
+----
+
+. Restore API Manager:
```
```suggestion
. Restore the API manager by using the following YAML:
```
Are we missing an apply command?

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
+  startTimestamp: "2024-09-24T05:59:39Z"
+----
+
+. Scale up the 3scale-operator:
```
```suggestion
. Scale up the 3scale Operator by running the following command:
```

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
++
+[source,terminal]
+----
+# oc scale deployment threescale-operator-controller-manager-v2 --replicas=1 -n threescale
```
```suggestion
$ oc scale deployment threescale-operator-controller-manager-v2 --replicas=1 -n threescale
```

ShaunaDiaz commented on modules/scale-operator-and-deploymentconfigs.adoc:
```diff
@@ -0,0 +1,59 @@
+----
+Use the URL from this output to log in to the 3scale admin.
+
+
```
```suggestion
```

ShaunaDiaz commented on modules/scale-operator-and-deploymentconfigs.adoc:
```diff
@@ -0,0 +1,59 @@
+.Verification
+
+* Verify if the existing data before creating the back up is available
+
```
delete all of these extra line spaces

ShaunaDiaz commented on modules/scale-operator-and-deploymentconfigs.adoc:
```diff
@@ -0,0 +1,59 @@
+
+.Verification
+
+* Verify if the existing data before creating the back up is available
```
```suggestion
* Verify that the existing data is available before trying to create a backup.
```
How? This is not a verification step if there's no actual instruction; this would just be an IMPORTANT admonition instead.

ShaunaDiaz commented on modules/scale-operator-and-deploymentconfigs.adoc:
```diff
@@ -0,0 +1,59 @@
+zync-3scale-provider-7r2nm   3scale-admin.apps.custom-cluster-name.openshift.com                           system-provider      http      edge/Redirect   None
+zync-3scale-provider-mjxlb   3scale.apps.custom-cluster-name.openshift.com                                 system-developer     http      edge/Redirect   None
+----
+Use the URL from this output to log in to the 3scale admin.
```
```suggestion

. Use the URL from this output to log in to the 3scale Operator as an administrator.
```

ShaunaDiaz commented on modules/scale-operator-and-deploymentconfigs.adoc:
```diff
@@ -0,0 +1,59 @@
+Done
+----
+
+. Get the `3scale-admin` route to login to the 3scale UI:
```
```suggestion
. Get the `3scale-admin` route to log in to the 3scale UI by running the following command:
```

ShaunaDiaz commented on modules/scale-operator-and-deploymentconfigs.adoc:
```diff
@@ -0,0 +1,59 @@
+
+.Procedure
+
+. Scale operator and DC:
```
Make this two separate steps.

ShaunaDiaz replied:
```suggestion
. Scale the Operator by running the following command:
```

ShaunaDiaz commented on modules/scale-operator-and-deploymentconfigs.adoc:
```diff
@@ -0,0 +1,59 @@
+----
+$ oc scale deployment threescale-operator-controller-manager-v2 --replicas=1 -n threescale
+----
++
```
. Scale the deployment configuration (DC) with a script by running the following commands:

ShaunaDiaz commented on modules/scale-operator-and-deploymentconfigs.adoc:
```diff
@@ -0,0 +1,59 @@
+    oc scale dc/$dc --replicas=1 -n threescale
+Done
+----
+
```
Are we missing the command to run the script here?

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
++
+[source,terminal]
+----
+# oc scale deployment threescale-operator-controller-manager-v2 --replicas=0 -n threescale
```
```suggestion
$ oc scale deployment threescale-operator-controller-manager-v2 --replicas=0 -n threescale
```

ShaunaDiaz replied:
Generally, we should not be suddenly from root. `oc` commands should be run by the cluster admin with their access, FWIW. Confirm with your SME and See https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#signs-and-symbols-in-code-blocks

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
+       creationTimestamp: null
+       name: s3-credentials
+stringData:
+  AWS_ACCESS_KEY_ID: something
```
```suggestion
  AWS_ACCESS_KEY_ID: <access_key> # <1>
```

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
+       name: s3-credentials
+stringData:
+  AWS_ACCESS_KEY_ID: something
+  AWS_SECRET_ACCESS_KEY: something
```
```suggestion
  AWS_SECRET_ACCESS_KEY: <aws_access_key> # <2>
```

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
+stringData:
+  AWS_ACCESS_KEY_ID: something
+  AWS_SECRET_ACCESS_KEY: something
+  AWS_BUCKET: something
```
```suggestion
  AWS_BUCKET: <aws_bucket> # <3>
```

apurvabhide17 replied:
Keeping this suggestion as it is, I've asked QE to find the real-life examples. I'll add its explanation using <1>, <2>, and such.

apurvabhide17 replied:
It is now fixed

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
+  AWS_REGION: us-east-1
+type: Opaque
+----
+
```
And define these fields...will the user always use AWS to spin up a cluster, or should there be a more generic example here?
<1> 
<2>
<3>

ShaunaDiaz commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,166 @@
+apiVersion: v1
+kind: Secret
+metadata:
+       creationTimestamp: null
```
```suggestion
       creationTimestamp: null
```
null?

ShaunaDiaz reviewed (commented):
Needs some work. I recommend getting a SME and QE reviews, then placing this back in peer review prior to submitting for merge.

apurvabhide17 commented:
> @apurvabhide17 All PRs pulled against the `main` branch must also go in 4.18. If this content cannot go in 4.18, you'll want to have it open against 4.17 and picked back.

Hello @ShaunaDiaz, I've updated the OCP version in the description of the PR. I confirmed it with the QE, and the changes can be merged to OCP 4.18 branch as well.
I've implemented most of the suggestions apart from the one's that I need to discuss again with the SMEs. Thank you for reviewing and adding helpful suggestions :)
The PR is on QE review and is shared with the Engineering team for further review. I'll share it again for the peer review once I get an ack from the SMEs.

apurvabhide17 commented on modules/backing-up-the-3scale-operator.adoc:
```diff
@@ -0,0 +1,131 @@
++
+[source,terminal]
+----
+$ oc create -f <backup>.yaml
```
fixed it in the code

stillalearner reviewed (approved):
looks good.

apurvabhide17 commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,178 @@
+.Example output:
+[source,terminal]
+----
+deploymentconfig.apps.openshift.io/apicast-production scaled
```
@stillalearner , could you please confirm if it should be deployment instead of "deploymentconfig" in the output?
Thanks!

mpershina commented on backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-and-restoring-3scale-using-oadp.adoc:
```diff
@@ -0,0 +1,36 @@
+
+With Red Hat 3scale API Management (APIM), you can manage your APIs for internal or external users. Share, secure, distribute, control, and monetize your APIs on an infrastructure platform built with performance, customer control, and future growth in mind.
+3scale components can be deployed on-premise, in the cloud, as a managed service, or in any combination based on your requirement.
+This document is using the non-service affecting approach to backup and restore 3scale on cluster storage using the {oadp-first} Operator. Here, either the source or destination cluster are the same or if not, both the clusters are using the same custom domain.
```
```suggestion
This document is using the non-service affecting approach to backup and restore 3scale on cluster storage by using the {oadp-first} Operator. Here, either the source or destination clusters are the same or both clusters are using the same custom domain.
```
Is it possible to get rid of "This document"? We're advised to avoid self-ref text as much as possible? Maybe there's a good way to rephrase this sentence?
Also, the second sentence is a bit hard to understand (for me, at least, hehe :-D). Any idea on how to rephrase it? (Not a requeitherirement, just a  suggestion)

apurvabhide17 replied:
Added it as a note - In this case, the non-service affecting approach is used to backup and restore 3scale on cluster storage by using the {oadp-first} Operator.

In addition to that, either use the same source or destination cluster. Otherwise, ensure that both the cluster are using the same custom domain

mpershina commented on modules/backing-up-the-3scale-operator.adoc:
```diff
@@ -0,0 +1,120 @@
+[id="backing-up-the-3scale-operator_{context}"]
+= Backing up the 3scale Operator
+
+Back up the operator resources, and secret and APIManager custom resources (CR). For more information, see link:https://docs.openshift.com/container-platform/4.17/backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-creating-backup-cr.html[Creating a Backup CR].
```
```suggestion
Back up the operator resources and secret and APIManager custom resources (CR). For more information, see link:https://docs.openshift.com/container-platform/4.17/backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-creating-backup-cr.html[Creating a Backup CR].
```
If this is an intro to the procedure, it should not contain directions, for example, "back up". It should rather be some info introducing what will be done through the procedure. Or, you could simply rephrase it if you don't have any useful info for the intro, for example:

> You can back up the following resources:
> * Operator resources
> * Secret and APIManager custom resources (CR)
> For more information, see link:https://docs.openshift.com/container-platform/4.17/backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-creating-backup-cr.html[Creating a Backup CR].

But I'd really suggest writing something useful. Like, for example, why would the user want to perform this procedure.

mpershina commented on modules/backing-up-the-3scale-operator.adoc:
```diff
@@ -0,0 +1,120 @@
+
+.Procedure
+
+. Back up the operator related resources such as `operatorgroup`, `namespaces`, and `subscriptions` by creating a YAML file with the following configurations: 
```
```suggestion
. Back up the operator related resources, such as `operatorgroup`, `namespaces`, and `subscriptions`, by creating a YAML file with the following configuration: 
```

mpershina commented on modules/backing-up-the-3scale-operator.adoc:
```diff
@@ -0,0 +1,120 @@
+$ oc create -f <backup>.yaml
+----
+
+. Back up the Secret CR by creating the following YAML file:
```
```suggestion
. Back up the Secret CR by creating a YAML file with the following configuration:
```
Just for the sake of parallelism :-)

apurvabhide17 replied:
thanks :)

mpershina commented on modules/backing-up-the-3scale-operator.adoc:
```diff
@@ -0,0 +1,120 @@
+$ oc create -f <backup-secret>.yaml
+----
+
+. Backup the APIManager CR by creating the following YAML file:
```
```suggestion
. Backup the APIManager CR by creating a YAML file with the following configuration:
```

mpershina commented on modules/backing-up-the-backend-redis-database.adoc:
```diff
@@ -0,0 +1,91 @@
+[id="backing-up-the-backend-redis-database_{context}"]
+= Backing up the backend Redis database
+
+You can backup the Redis database by adding the annotations mentioned in this procedure and by adding the resources using the `includedResources` parameter, which must be backed up.
```
```suggestion
You can back up the Redis database by adding the annotations mentioned in the following procedure and by adding the resources by using the `includedResources` parameter, which must be backed up.
```

mpershina commented on modules/backing-up-the-backend-redis-database.adoc:
```diff
@@ -0,0 +1,91 @@
+
+.Procedure
+
+. Edit the annotations on the `backend-redis` deployment by running the following command:
```
```suggestion
. Edit the annotations on the `backend-redis` deployment:
```
" by running the following command" can really be dropped.

apurvabhide17 replied:
I agree but it was suggested in the OCP review

mpershina commented on modules/backing-up-the-backend-redis-database.adoc:
```diff
@@ -0,0 +1,91 @@
+  ttl: 720h0m0s
+----
+
+. Backup the Redis database using the following command:
```
```suggestion
. Back up the Redis database:
```
Same  here, no need for "by running the following command"

mpershina commented on modules/backing-up-the-backend-redis-database.adoc:
```diff
@@ -0,0 +1,91 @@
+
+.Verification
+
+* Verify if the Redis backup is completed: 
```
```suggestion
* Verify that the Redis backup is completed: 
```

mpershina commented on modules/backing-up-the-mysql-database.adoc:
```diff
@@ -0,0 +1,144 @@
+
+.Procedure
+
+. Create a YAML file with the following configurations for adding an additional PVC:
```
```suggestion
. Create a YAML file with the following configuration for adding an additional PVC:
```

mpershina commented on modules/backing-up-the-mysql-database.adoc:
```diff
@@ -0,0 +1,144 @@
+  volumeMode: Filesystem
+----
+
+. Create the additional PVC by running the following command: 
```
```suggestion
. Create the additional PVC: 
```

mpershina commented on modules/backing-up-the-mysql-database.adoc:
```diff
@@ -0,0 +1,144 @@
+            claimName: example-claim <1>
+    ...
+----
+<1> PVC, which contains the dumped data
```
```suggestion
<1> PVC, which contains the dumped data.
```

mpershina commented on modules/backing-up-the-mysql-database.adoc:
```diff
@@ -0,0 +1,144 @@
+  snapshotMoveData: false
+  ttl: 720h0m0s
+----
+<1> Data is backed up in this directory.
```
```suggestion
<1> A directory where the data is backed up.
```

mpershina commented on modules/backing-up-the-mysql-database.adoc:
```diff
@@ -0,0 +1,144 @@
+<1> Data is backed up in this directory.
+<2> The included resources will be backed up.
+
+. Backup the `mysql` database:
```
```suggestion
. Back up the `mysql` database:
```

mpershina commented on modules/backing-up-the-mysql-database.adoc:
```diff
@@ -0,0 +1,144 @@
+
+.Verification
+
+* Verify if the `mysql` backup is completed: 
```
```suggestion
* Verify that the `mysql` backup is completed: 
```

mpershina commented on modules/backing-up-the-mysql-database.adoc:
```diff
@@ -0,0 +1,144 @@
+
+.Next step
+
+* Backup the backend Redis database.
```
```suggestion
* Back up the back-end Redis database.
```
IBM SG for back-end as an adjective.

mpershina commented on modules/creating-the-data-protection-application.adoc:
```diff
@@ -0,0 +1,67 @@
+[id="creating-the-data-protection-application_{context}"]
+= Creating the Data Protection Application
+
+Create a Data Protection Application (DPA) custom resource (CR) for 3scale. For more information on DPA, see link:https://docs.openshift.com/container-platform/4.17/backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.html#oadp-installing-dpa_installing-oadp-aws[Installing the Data Protection Application].
```
Again, we should not use directions for procedure's intro. It should rather be some reason why the user would want to perform the steps. 

If there's really nothing to write about, then it should be at least written without the directions, for example:

> You can create a Data Protection Application (DPA) custom resource (CR) for 3scale. For more information on DPA, see link:https://docs.openshift.com/container-platform/4.17/backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.html#oadp-installing-dpa_installing-oadp-aws[Installing the Data Protection Application].

mpershina commented on modules/restoring-the-backend-redis-database.adoc:
```diff
@@ -0,0 +1,82 @@
+//included in backing-up-and-restoring-3scale-using-oadp.adoc assembly
+
+[id="restoring-the-backend-redis-database_{context}"]
+= Restoring the backend Redis database
```
```suggestion
= Restoring the back-end Redis database
```
IBM SG

mpershina commented on modules/restoring-the-backend-redis-database.adoc:
```diff
@@ -0,0 +1,82 @@
+  restorePVs: true
+----
+
+. Restore the Redis database by using the following command:
```
```suggestion
. Restore the Redis database:
```

mpershina commented on modules/restoring-the-backend-redis-database.adoc:
```diff
@@ -0,0 +1,82 @@
+
+.Verification 
+
+* Verify that the `PodVolumeRestore` restore is completed by running the following command:
```
```suggestion
* Verify that the `PodVolumeRestore` restore is completed:
```

mpershina commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,178 @@
+[id="restoring-the-mysql-database_{context}"]
+= Restoring the mysql database
+
+The restoration of the `mysql` database recreates the `Pod`, `ReplicationController`, and deployment objects, plus the additional persistent volumes (PVs) and associated persistent volume claims (PVCs). 
+It restores the `mysql` dump, which the `example-claim` persistent volume claim (PVC) contains.
```
```suggestion
Restoration of the `mysql` database causes the following processes:

* Recreation of the `Pod`, `ReplicationController`, and deployment objects.
* Recreation of the additional persistent volumes (PVs) and associated persistent volume claims (PVCs). 
* Restoration of the `mysql` dump, which the `example-claim` persistent volume claim (PVC) contains.
```
A suggestion for better readability. Feel free to rephrase even more ;-)

apurvabhide17 replied:
Thanks, that's an awesome suggestion :)

mpershina commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,178 @@
+deploymentconfig.apps.openshift.io/zync-que scaled
+----
+
+. Delete the `system-mysql` deployment object by running the following command:
```
```suggestion
. Delete the `system-mysql` deployment object:
```

mpershina commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,178 @@
+----
+<1> The data is restored from this path.
+
+. Restore the `mysql` database with by running the following command:
```
```suggestion
. Restore the `mysql` database:
```

mpershina commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,178 @@
+
+.Verification 
+
+. Verify that the `PodVolumeRestore` is completed by running the following command:
```
```suggestion
. Verify that the `PodVolumeRestore` is completed:
```

mpershina commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,178 @@
+restore-mysql-z7x7l     threescale   system-mysql-2-kjkhl    kopia           example-claim           Completed   380415       380415      40m
+----
+
+. Verify that the additional persistent volume claim (PVC) has also been restored by running the following command:
```
```suggestion
. Verify that the additional persistent volume claim (PVC) has been restored:
```

mpershina commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,178 @@
+
+.Next step
+
+* Restore the backend Redis database.
```
```suggestion
* Restore the back-end Redis database.
```
IBM SG for the adjective

mpershina commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,171 @@
+
+.Procedure
+
+. Delete the 3scale Operator custom resource definitions (CRDs) along with the "threescale" namespace by running the following command:
```
```suggestion
. Delete the 3scale Operator custom resource definitions (CRDs) along with the "threescale" namespace:
```

mpershina commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,171 @@
++
+[IMPORTANT]
+====
+If you are restoring the database on a different cluster than you backed up, ensure that {oadp-short} is installed and configured with `nodeAgent` enabled on the destination cluster as it was on to the source cluster.
```
A bit hard to understand..Is there a better way to say the same? 
I'd also advise to define `nodeAgent`.

apurvabhide17 replied:
Added it to prerequisite - Ensure that you are restoring the database on the same cluster, where it was backed up. If it is on a different cluster, install and configure {oadp-short} with `nodeAgent` enabled on the destination cluster as it was on to the source cluster.

mpershina commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,171 @@
+  itemOperationTimeout: 4h0m0s
+----
+
+. Restore the 3scale Operator by running the following command:
```
```suggestion
. Restore the 3scale Operator:
```

mpershina commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,171 @@
+$ oc create -f restore.yaml
+----
+
+. Manually create the `s3-credentials` Secret by running the following command:
```
```suggestion
. Manually create the `s3-credentials` Secret:
```

mpershina commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,171 @@
+EOF
+----
+
+. Scale down the 3scale Operator by running the following command:
```
```suggestion
. Scale down the 3scale Operator:
```

mpershina commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,171 @@
+  itemOperationTimeout: 4h0m0s
+----
+
+. Restore the Secrets by running the following command:
```
```suggestion
. Restore the Secrets:
```

mpershina commented on modules/scale-up-the-3scale-operator-and-deployment.adoc:
```diff
@@ -0,0 +1,71 @@
+//included in backing-up-and-restoring-3scale-using-oadp.adoc assembly
+
+[id="scale-up-the-3scale-operator-and-deployment_{context}"]
+= Scale up the 3scale operator and deployment
```
```suggestion
= Scaling up the 3scale operator and deployment
```
The name should contain the -ing verb as per our rules for procedures.

apurvabhide17 replied:
Good catch, missed adding a gerund for this module :)

mpershina commented on modules/scale-up-the-3scale-operator-and-deployment.adoc:
```diff
@@ -0,0 +1,71 @@
+[id="scale-up-the-3scale-operator-and-deployment_{context}"]
+= Scale up the 3scale operator and deployment
+
+Scale up the 3scale operator and any deployment that were manually scaled down. After a few minutes, 3scale installation should be fully functional, and its state should match the backed-up state.
```
```suggestion
You can scale up the 3scale operator and any deployment that was manually scaled down. After a few minutes, 3scale installation should be fully functional, and its state should match the backed-up state.
```

mpershina commented on modules/scale-up-the-3scale-operator-and-deployment.adoc:
```diff
@@ -0,0 +1,71 @@
+
+.Prerequisites
+
+* Ensure that there are no deployments scaled up already, or any extra pods are running. 
```
```suggestion
* Ensure that there are no scaled up deployments or no extra pods are running. 
```

mpershina commented on modules/scale-up-the-3scale-operator-and-deployment.adoc:
```diff
@@ -0,0 +1,71 @@
+$ oc get deployment -n threescale
+----
+
+. Scale up the deployments using the following script:
```
```suggestion
. Scale up the deployments by running the following script:
```

mpershina commented on modules/scale-up-the-3scale-operator-and-deployment.adoc:
```diff
@@ -0,0 +1,71 @@
+$ ./scaledeployment.sh
+----
+
+. Get the `3scale-admin` route to log in to the 3scale UI by running the following command:
```
```suggestion
. Get the `3scale-admin` route to log in to the 3scale UI:
```

mpershina commented on _topic_maps/_topic_map.yml:
```diff
@@ -3575,6 +3575,11 @@ Topics:
+  - Name: OADP and 3scale
+    Dir: oadp-3scale
+    Topics:
+    - Name: Backing up and restoring 3scale using OADP
```
```suggestion
    - Name: Backing up and restoring 3scale by using OADP
```

apurvabhide17 commented:
The changes are approved by QE and SME. Adding a final peer review request, thanks in advance.

/label peer-review-needed

apurvabhide17 commented:
The changes are approved by QE and SME. Adding a final peer review request, thanks in advance.

/label peer-review-needed

bergerhoffer commented:
/label peer-review-in-progress

bergerhoffer commented on modules/restoring-the-backend-redis-database.adoc:
```diff
@@ -0,0 +1,84 @@
+
+.Verification 
+
+* Verify that the `PodVolumeRestore` restore is completed:
```
In an earlier module, you just said this:

```suggestion
* Verify that the `PodVolumeRestore` is completed:
```
Not sure which is correct, but I'd pick one and be consistent between the two places.

bergerhoffer reviewed (commented):
Lots of little nits - please be sure to review them and make the changes across all the files, since I didn't point out all instances of some of the suggestions.

Let me know if you have any questions on any of the feedback!

bergerhoffer commented:
/label peer-review-done
/remove-label peer-review-needed
/remove-label peer-review-in-progress

apurvabhide17 commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,173 @@
++
+[source,terminal]
+----
+$ oc apply -f - <<EOF
```
It was suggested by the SME for this file but I can check it again.

apurvabhide17 commented on backup_and_restore/application_backup_and_restore/oadp-3scale/backing-up-and-restoring-3scale-by-using-oadp.adoc:
```diff
@@ -0,0 +1,42 @@
+
+include::modules/backing-up-the-backend-redis-database.adoc[leveloffset=+1]
+
+include::modules/restoring-the-secrets-and-apimanager.adoc[leveloffset=+1]
```
Thanks for suggesting this. I'll add it later :)

apurvabhide17 commented on backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-scheduling-backups-doc.adoc:
```diff
@@ -32,7 +32,7 @@ For example, if a backup of a namespace typically takes 10 minutes, do not sched
 ----
 +
-.Example output
+.Example output:
```
Not edited in this PR.

kcarmichael08 replied:
I don't understand what "not edited in this PR" means since it's showing edited in this PR. If you didn't make changes to this file, then I think it needs to be removed from the PR (especially since it's a wrong change, adding the colon).

apurvabhide17 replied:
I've fixed it, it was by mistake that this file was edited.

apurvabhide17 commented on modules/restoring-the-secrets-and-apimanager.adoc:
```diff
@@ -0,0 +1,173 @@
+
+.Prerequisites
+
+. You backed up the 3scale Operator.
```
Since it must be executed in the sequence, prerequisites are having an ordered list instead of bullets.
@bergerhoffer, is there any recommendation for Prerequisites for OCP for using ordered or unordered list?
Thanks in advance!

apurvabhide17 commented on modules/creating-the-data-protection-application.adoc:
```diff
@@ -0,0 +1,67 @@
+          name: cloud-credentials
+----
+
+. Create a DPA CR:
```
I've fixed it all the files

apurvabhide17 commented on modules/restoring-the-mysql-database.adoc:
```diff
@@ -0,0 +1,181 @@
++
+[source,terminal]
+----
+$ cat ./scaledowndeployment.sh
```
I've fixed the command.

apurvabhide17 commented on modules/backing-up-the-3scale-operator.adoc:
```diff
@@ -0,0 +1,120 @@
+  - ts-dpa-1
+----
+
+. Create the APIManager CR:
```
Correct way to write it is - APIManager. Adding the readme - https://github.com/3scale/3scale-operator/blob/master/doc/apimanager-reference.md.
Referring secrets and APIManager without backticks.

apurvabhide17 commented:
/label merge-review-needed

kcarmichael08 commented on modules/scaling-up-the-3scale-operator-and-deployment.adoc:
```diff
@@ -0,0 +1,60 @@
+zync-3scale-provider-mjxlb   3scale.apps.custom-cluster-name.openshift.com                                 system-developer     http      edge/Redirect   None
+----
++
+In this eample, `3scale-admin.apps.custom-cluster-name.openshift.com` is the 3scale-admin URL.
```
```suggestion
In this example, `3scale-admin.apps.custom-cluster-name.openshift.com` is the 3scale-admin URL.
```

apurvabhide17 replied:
oh, my bad, thanks for pointing.

kcarmichael08 reviewed (commented):
This looks really good! Just a couple things - I left a question about the change adding the colon to example, and there is a typo. I think if you could implement Andrea's suggested reorg in the future, that would be cool. Nice job!

apurvabhide17 commented:
> This looks really good! Just a couple things - I left a question about the change adding the colon to example, and there is a typo. I think if you could implement Andrea's suggested reorg in the future, that would be cool. Nice job!

Thanks @kcarmichael08, I've fixed all the minor issues. I'll do the re-org as Andrea suggested after this PR is merged along with basic minor updates to two commands - **oc get deployment -n threescale** and **oc edit deployment system-mysql -n threescale**. I've already pinged my SME for these commands.
For now, iss it possible to merge this PR?

apurvabhide17 commented on modules/backing-up-the-mysql-database.adoc:
```diff
@@ -0,0 +1,144 @@
+----
+$ oc edit deployment system-mysql -n threescale
+----	
++
```
The oc edit command opens the YAML for that deployment resource as mentioned in the step description and it is not a file.

kcarmichael08 commented:
/cherrypick enterprise-4.14

kcarmichael08 commented:
/cherrypick enterprise-4.15

kcarmichael08 commented:
/cherrypick enterprise-4.17

kcarmichael08 commented:
/cherrypick enterprise-4.18
